make_matcher_backend/app/models/concerns/geocode.rb

```
@@ -0,0 +1,66 @@
+module Geocode
+  require 'uri'
+  require 'net/http'
+  require 'json'
+
+  API_KEY = ENV['GEOCODE_KEY'] || Rails.application.credentials.geocode[:api_key]
+  GEOCODE_URL = "https://geocode.maps.co/search?api_key=#{API_KEY}&country=US&postalcode=".freeze
+  TIMEZONE_URL = 'https://www.timeapi.io/api/TimeZone/coordinate?'.freeze
+  UTC_OFFSET_URL = 'https://www.timeapi.io/api/TimeZone/zone?timeZone='.freeze
+
+  def self.coordinate(zip_code)
+    uri = URI("#{GEOCODE_URL}#{zip_code}")
+    res = Net::HTTP.get_response(uri)
+    return 'There was an error' unless res.is_a?(Net::HTTPSuccess)
```

probably cleaner to throw a custom exception instead of returning `nil` or string error messages

---

Anything you have in mind here?
I didn't want to have the before_save callback prevent an update from happening, so I was thinking of just logging the exception and moving on.

---

I would consider using `after_save` instead, but you can also catch the exception in the `locate` callback method as well

---

also
```
  rescue StandardError => e
    raise e
```
is unnecessary unless you're doing something like logging

---

I think I'm good with the latest commits.   Update still happens with the rescue and it will try to locate again on the next save if there are no calculated coordinate or utc_offset values.